lib/cloudantBaseService.ts

```diff
@@ -177,31 +177,33 @@ export default abstract class CloudantBaseService extends BaseService {
       // Extract operation id
       const analyticsHeader =
         parameters.defaultOptions.headers['X-IBMCloud-SDK-Analytics'];
-      for (const element of analyticsHeader.split(';')) {
-        if (element.startsWith('operation_id')) {
-          [, operationId] = element.split('=');
-        }
-      }
+      [, operationId] = analyticsHeader
+        .split(';')
+        .find((element) => element.startsWith('operation_id'))
+        .split('=');
       // Check if operation id exists in rulesByOperation object
       if (
         operationId != null &&
         Object.keys(rulesByOperation).includes(operationId)
       ) {
-        for (const rule of rulesByOperation[operationId]) {
-          // get the path segment e.g. doc_id from the response's path object
```

Comment:
I think the removal of this comment is unintentional. Now it should be placed under `const violatedRules = rulesByOperation[operationId].filter((rule) => {`

Reply:
Right, placed this back: 06afc4a